spec/controllers/submissions_controller_spec.rb

```diff
@@ -958,6 +962,23 @@
             expect(response).to render_template(partial: 'shared/_poll_job')
           end
 
+          it 'should succeed if it is before the section due date and collect_current is selected' do
+            @assessment_section_properties.update!(due_date: 1.week.from_now)
+            allow(Assignment).to receive_message_chain(
+              :includes, :find
+            ) { @assignment }
+            expect_any_instance_of(SubmissionsController).not_to receive(:flash_now).with(:error, anything)
```

Comment:
This expect statement is checking whether flash_now is called. It's more robust to test if anything is added to the flash hash at any point. Can we move this test to after the `post_as` call and instead change the test to something like:

```ruby
expect(flash[:error]).to be_nil
```

There are examples of this in a bunch of controller specs, if you search for `expect(flash[:error])` in the source code it should show a bunch of examples